SwiftCharts/Views/ChartAreasView.swift

```diff
@@ -11,92 +11,68 @@ import UIKit
 open class ChartAreasView: UIView {
 
     fileprivate let animDuration: Float
-    fileprivate let colors: [UIColor]
+    fileprivate var colors: [UIColor]
     fileprivate let animDelay: Float
+    fileprivate let addContainerPoints: Bool
     
-    public init(points: [CGPoint], frame: CGRect, colors: [UIColor], animDuration: Float, animDelay: Float) {
+    public init(points: [CGPoint], frame: CGRect, colors: [UIColor], animDuration: Float, animDelay: Float, addContainerPoints: Bool, pathGenerator: ChartLinesViewPathGenerator) {
         self.colors = colors
         self.animDuration = animDuration
         self.animDelay = animDelay
+        self.addContainerPoints = addContainerPoints
         
         super.init(frame: frame)
 
         backgroundColor = UIColor.clear
-        show(path: generateAreaPath(points: points))
+        show(path: generateAreaPath(points: points, pathGenerator: pathGenerator))
     }
     
     required public init(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
-    fileprivate func generateAreaPath(points: [CGPoint]) -> UIBezierPath {
-        
-        let progressline = UIBezierPath()
-        progressline.lineWidth = 1.0
-        progressline.lineCapStyle = .round
-        progressline.lineJoinStyle = .round
-        
-        if let p = points.first {
-            progressline.move(to: p)
-        }
-        
-        for i in 1..<points.count {
-            let p = points[i]
-            progressline.addLine(to: p)
-        }
-        
-        progressline.close()
-        
-        return progressline
+    
+    fileprivate func generateAreaPath(points: [CGPoint], pathGenerator: ChartLinesViewPathGenerator) -> UIBezierPath {
+        return pathGenerator.generateAreaPath(points: points, lineWidth: 1.0)
     }
     
     fileprivate func show(path: UIBezierPath) {
         guard let firstColor = colors.first else {
             print("WARNING: No color(s) used for ChartAreasView")
             return
         }
-        let areaLayer = CAShapeLayer()
-        areaLayer.lineJoin = kCALineJoinBevel
-        areaLayer.fillColor   = firstColor.cgColor
-        areaLayer.lineWidth   = 2.0
-        areaLayer.strokeEnd   = 0.0
+        
         if colors.count == 1 {
```

Comment:
Looks in general better than the old implementation - anyway, I'd maybe a short comment here explaining why you have to re-add the color, isn't immediately clear that it's because of the gradient.

Reply:
Yeah a comment here could be useful. I added a brief explanation.